variables.tf

```diff
@@ -244,6 +244,8 @@ variable "roles" {
     path                 = string       # Defaults to 'var.role_path' if variable is set to null
     desc                 = string       # Defaults to 'var.role_desc' if variable is set to null
     trust_policy_file    = string       # Path to file of trust/assume policy
```

Comment:
Why not simply make the `trust_policy_file` a template file and get rid of the additional `trust_policy_body`. Then you could use optional `trust_policy_vars` with the already existing file variable

Reply:
I’d prefer to stick with the separation of concerns, in this case. WDYT?

Reply:
I'd stick with `trust_policy_file` and `trust_policy_vars`. This makes it much cleaner.

Downside of having `_file` and `_body` is that from the outside of the module you don't really see what the difference is and that only one of them can be used and also not what the precedence is.

Having just one, omits the user decision and makes it more clear.